src/pages/patientView/PatientViewPage.tsx

```diff
@@ -90,11 +95,52 @@ export interface PatientViewUrlParams extends QueryParams {
     sampleId?: string;
 }
 
+/*
+ * The wrapper and the inner component were instituted so that
+ * because we needed a component to remount (and create a new patient view store)
+ * whenever params changed
+ * There was a bug when user navigated between cases in a cohort
+ * Safest/easiest solution under cicumstances is just to reinstantiate everything
+ */
+
 @inject('routing', 'appStore')
 @observer
 export default class PatientViewPage extends React.Component<
     IPatientViewPageProps,
     {}
+> {
+    cohortIds: string[] | undefined;
+
+    constructor(props: IPatientViewPageProps) {
+        super(props);
+
+        const postData = getBrowserWindow().clientPostedData;
+
+        const urlData = getNavCaseIdsCache();
+
+        if (postData && postData.navCaseIds) {
+            this.cohortIds = buildCohortIdsFromNavCaseIds(postData.navCaseIds);
+            getBrowserWindow().clientPostedData = null;
+        } else if (urlData) {
+            this.cohortIds = urlData;
+        }
+    }
+
+    render() {
+        return (
+            <PatientViewPageInner
+                key={`${this.props.routing.query.caseId}-${this.props.routing.query.studyId}-${this.props.routing.query.sampleId}`}
+                {...this.props}
+                cohortIds={this.cohortIds}
+            />
+        );
+    }
+}
```

Comment:
I would add an empty line here
```suggestion
}

```